### PR TITLE
Add display name to gateway form

### PR DESCRIPTION
**Pull request by gherceg**
[SMS Pricing on website update](https://dimagi-dev.atlassian.net/browse/SAAS-11388)
##### SUMMARY

Not directly related to the ticket, but Amy mentioned in the comments that the Apposit gateway was missing a label on the SMS pricing page. This is because the gateway backend is missing a display name, so adding this to easily view/edit display names on gateway backends moving forward. I cleaned up some flake warnings as well. 

Still new to forms in django so let me know if I missed anything.

### CODE REVIEW

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/sms/forms.py [hunk 1]
```diff
@@ -1001,6 +1005,12 @@ def clean_name(self):
 
         return value
 
+    def clean_display_name(self):
+        value = self.cleaned_data.get("display_name")
+        if re.compile(r"\d").search(value) is not None:
+            raise ValidationError(_("Display name may not contain any numbers."))
```
Comment (millerdev) [inline corehq/apps/sms/forms.py:1011]: I'm curious about this requirement. The validation logic will allow any character other than numeric digits. Does that mean special/punctuation characters are allowed? If yes, why not numbers? Or maybe it should only allow ASCII characters?

Comment (gherceg) [reply corehq/apps/sms/forms.py:1011]: Good point. Given that this is an internal tool I'm tempted to say no validation is even necessary. I don't have a strong opinion on this though so if only allowing ASCII characters seems better I'll make that change.

Comment (millerdev) [reply corehq/apps/sms/forms.py:1011]: > Given that this is an internal tool I'm tempted to say no validation is even necessary.

I'm pro dropping the validation. This seems like the kind of thing someone might read later and say "Why was it done that way?" A comment explaining the "why" might be nice if the validation is kept.

Comment (gherceg) [reply corehq/apps/sms/forms.py]: Removed validation.

**Pull request merged**